### PR TITLE
Minor MotionMaster refactor

### PR DESCRIPTION
**Pull request by chaodhib**
**Changes proposed:**
a bit of clean up to MotionMaster::UpdateMotion()

**Target branch(es):** 3.3.5/master

- [x] 3.3.5

**Issues addressed:** Does not change any behavior. This is just code clean up.

### CODE REVIEW

**Comment by ccrs** [comment]
Merging in a couple mins, if you dont want to include anything else.

**Comment by chaodhib** [comment]
Nop, that's all I have for now. Thanks

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -467,6 +467,7 @@ void Unit::Update(uint32 p_time)
 
     UpdateSplineMovement(p_time);
     i_motionMaster->UpdateMotion(p_time);
+    UpdateUnderwaterState(GetMap(), GetPositionX(), GetPositionY(), GetPositionZ());
```
Comment (Shauren) [inline src/server/game/Entities/Unit/Unit.cpp]: expensive...

Comment (chaodhib) [inline src/server/game/Entities/Unit/Unit.cpp]: This method was called in UpdateMotion() (the line above). So it was called on each Unit::Update() call anyway. And putting this call here is better than in MotionMaster::UpdateMotion() imo.

Comment (Takenbacon) [inline src/server/game/Entities/Unit/Unit.cpp]: Seems like it's pretty pointless considering it's already called in Unit::UpdatePosition

Comment (chaodhib) [inline src/server/game/Entities/Unit/Unit.cpp]: I removed it from Unit::UpdatePosition and moved it in Unit::Update. This method better belong here than in the MotionMaster.

File: src/server/game/Movement/MotionMaster.cpp [hunk 2]
```diff
@@ -88,37 +88,14 @@ void MotionMaster::UpdateMotion(uint32 diff)
     ASSERT(!empty());
 
     _cleanFlag |= MMCF_UPDATE;
-    if (!top()->Update(_owner, diff))
-    {
-        _cleanFlag &= ~MMCF_UPDATE;
+    bool IsMoveGenUpdateSuccess = top()->Update(_owner, diff);
```
Comment (Shauren) [inline src/server/game/Movement/MotionMaster.cpp]: you call it a refactor but dont follow codestyle (local variable naming)...

Comment (chaodhib) [inline src/server/game/Movement/MotionMaster.cpp]: fixed

**Pull request merged**